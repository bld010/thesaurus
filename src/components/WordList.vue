<template>
  <section>
    <ul>
      <li v-for="(synonym, index) in synonyms" :key="index" tabindex=0 @click="emitSearchTermToApp(synonym)" @keyup="keyUpOnSynonym(synonym)">
        {{synonym}}
        </li>
    </ul>
  </section>
</template>

<script>


export default {
  name: 'WordList',
  props: ['synonyms'],
  methods: {
    emitSearchTermToApp: function(synonym) {
      event.target.blur()
      this.$emit('onSynonymClick', synonym)
    },
    keyUpOnSynonym: function(synonym) {
      if (event.key == "Enter") {
        this.emitSearchTermToApp(synonym)
      }
    }
  }
}

</script>

<style scoped>

section {
  overflow-y: scroll;
  min-height: 200px;
  max-height: 500px;
  width: 400px;
  border: 1px solid #039B77;
  margin: auto;
}

  ul {
    /* min-height: 200px; */
    list-style: none;
    /* max-width: 400px; */
    margin: auto;
    padding: 20px 0px 20px 40px;
    /* max-height: 400px; */
    /* height: 400px; */
    column-count: 2;
    column-width: 50%;
    overflow-y: scroll;
  }

  li {
    margin: 10px;
    padding: 5px;
    width: 50%;
  }

  li:hover {
    cursor: pointer
  }


  
</style>