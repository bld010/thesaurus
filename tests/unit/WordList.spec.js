import WordList from '../../src/components/WordList.vue';
import { shallowMount } from '@vue/test-utils';
import '@vue/cli-plugin-unit-jest'

describe('WordList', () => {
  it('should match the snapshot', () => {
    const wrapper = shallowMount(WordList);

    expect(wrapper).toMatchSnapshot();
  })
})