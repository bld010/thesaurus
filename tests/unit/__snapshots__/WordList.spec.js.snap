// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordList should match the snapshot 1`] = `
<section>
  <ul></ul>
</section>
`;
